### PR TITLE
Circle Class Submission

### PR DESCRIPTION
**Pull request by ahwall78**

### CODE REVIEW

File: students/ahwall/Lesson08HomeworkGitIn/Circle.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+# Step 4
+    @property
+    def area(self):
+        return math.pi * (self.radius ** 2)
+
+# Step 5
+    # @classmethod
+    # def from_diameter(cls, diameter):    FIX
+    #     return cls.diameter
```
Comment (hosungs) [inline students/ahwall/Lesson08HomeworkGitIn/Circle.py]: `cls(...)` will give you an instance of the `cls` class type.

File: students/ahwall/Lesson08HomeworkGitIn/Circle.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    def __str__(self):
+        return "Circle with radius: {:6f}".format(self.radius)
+
+    def __repr__(self):
+        return "Circle({})".format(self.radius)
+
+# Step 7
+    def __add__(self, other):
+        add_cir_rad = Circle(self.radius + other.radius)
+        return eval(repr(add_cir_rad))
```
Comment (hosungs) [inline students/ahwall/Lesson08HomeworkGitIn/Circle.py]: Why use `eval(repr(...))`? Just `return add_cir_rad`!

File: students/ahwall/Lesson08HomeworkGitIn/Circle.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    def __mul__(self, multiplier):  # work on flipped multiplier
+        mul_cir_rad = Circle(self.radius * multiplier)
+        return eval(repr(mul_cir_rad))
```
Comment (hosungs) [inline students/ahwall/Lesson08HomeworkGitIn/Circle.py]: Ditto

**Comment by hosungs** [comment]
Oops, another merge conflict... Strange... Anyway, that's why I can't close/merge this PR. Not 100% sure if this is going to work, but try the following and let us know if it doesn't work. We'll need to work this out in person in class or at office hours.

```
$ git pull -X theirs upstream master
$ git push origin master
```

**Comment by PythonCHB** [comment]
we can't merge this, as there are a lot of conflicts. 

These seem to be conflicts in files that you should not have changed -- it's very easy to accidentally change things:

```
examples/Session07/refactor.py
examples/Session08/fraction.py
examples/Session08/test_fraction.py 
```
These examples should be only changed by the instructors -- but we asked you to start with them and build on them, so it's easy to change stuff by accident. Ideally, you could copy them to your own dir, and then start making changes.

```
notes_for_class/source/index.rst
notes_for_class/source/session08.rst 
```

These are also only supposed to be touched by the instructors -- unless you are suggesting corrections, or ... in which case, do make it clear that you are doing that.

```
students/mathewcm/journal.md
students/mathewcm/lesson06/mailroom.py
students/mathewcm/lesson06/test_mailroom.py
students/tammyd/lesson03/strformat_lab.py 
```

these are files in other students' dirs -- it's possible that you haven't changed anything in these, but that the merge confilicts happened somewhere else.

So how to clean this all up? Here is a page about some of that:

https://uwpce-pythoncert.github.io/PythonCertDevel/supplemental/dev_environment/git_hints.html#backing-out-a-change

But I'm going to see what I can do to clean this up on this end.

Stay tuned

**Comment by PythonCHB** [comment]
So this is a bit of a mess.

I've added your Circle code to the main repo by hand, and am closing this PR.

We'll see what happens when you open a new one!

**Pull request closed without merging**